crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -253,11 +253,34 @@ where
             metrics: Metrics::default(),
         };
 
-        if let Some(target) = target {
+        let maybe_pipeline_target = match target {
+            // Provided target always takes precedence.
+            target @ Some(_) => target,
+            None => {
+                // If no target was provided, check if the stages are congruent - check if the
+                // checkpoint of the last stage matches the checkpoint of the first.
+                let headers_stage_checkpoint = this
+                    .blockchain
+                    .get_stage_checkpoint(StageId::Headers)?
```

Comment:
this sounds correct to me,

it'd be useful to add a few more docs here, when this can happen and what the impact of this triggering the pipeline with the header hash is